packages/graphql-key-transformer/src/KeyTransformer.ts

```diff
@@ -74,6 +76,31 @@ export class KeyTransformer extends Transformer {
     );
   }
 
+  public before = (ctx: TransformerContext): void => {
+    // generate a Map for the (resolver , VTLExpression)
+    const resolverMap: Map<string, string> = new Map();
+    ctx.metadata.set(ResourceConstants.SNIPPETS.SyncResolverKey, resolverMap);
+  };
+
+  public after = (ctx: TransformerContext): void => {
+    // construct sync VTL code
+    const resolverMap = ctx.metadata.get(ResourceConstants.SNIPPETS.SyncResolverKey);
+    if (resolverMap.size) {
+      resolverMap.forEach((syncVTLContent, resource) => {
+        if (syncVTLContent) {
+          resource.Properties.RequestMappingTemplate = '';
+          resource.Properties.RequestMappingTemplate = joinSnippets([
+            print(generateSyncResolverInit()),
+            syncVTLContent,
+            print(setSyncQueryFilterSnippet()),
+            print(setSyncKeyExpressionForHashKey(ResourceConstants.SNIPPETS.ModelQueryExpression)),
+            print(setSyncKeyExpressionForRangeKey(ResourceConstants.SNIPPETS.ModelQueryExpression)),
```

Comment:
Do we need to add the Hash/Range Key Expressions on every Resolver Map?

Reply:
yeah , we need to add hash/range key expressions when @key directive is present